core/src/smartcontracts/isi/account.rs

```diff
@@ -298,18 +299,8 @@ pub mod isi {
             // Check if account exists
             wsv.account_mut(&account_id)?;
 
-            if !wsv
-                .permission_token_definitions()
-                .contains_key(&permission.definition_id)
-            {
-                return Err(FindError::PermissionTokenDefinition(permission.definition_id).into());
-            }
             if !wsv.remove_account_permission(&account_id, &permission) {
-                return Err(FindError::PermissionToken(PermissionTokenFindError {
-                    account_id: account_id.clone(),
-                    permission_token_id: permission.definition_id,
-                })
-                .into());
+                return Err(FindError::PermissionToken(permission.definition_id).into());
             }
```

Comment:
to tell you the truth error should never be returned from this function call

Reply:
let's put it like this:

1. This function is only called by the validator
2. Permission token definitions are owned by the validator and provided to Iroha during migration. There is no point in checking if a permission token definition (that was received from validator) exists in the table that we store in Iroha because we got this table from validator itself. These checks belong to the validator

Reply:
> This function is only called by the validator

Wait, why? It's `Revoke<Account, PermissionToken>`. Anybody can call it if it's not prohibited by Validator

Reply:
any call would go through the executor. Executor owns the permission tokens and should validate that the one it received exists before calling executing `Revoke`. Iroha can't really be certain that a permission token exists.

Reply:
If so, I think you should add these checks into our `default` validator and `validator_with_admin`

Reply:
but isn't this checked already. If you look at `visit_revoke_account_permission` or `visit_grant_account_permission` I believe we construct the permission token there. Can you confirm this?

Reply:
Indeed, it's done with `is_owned_by()`